std/algorithm/iteration.d

```diff
@@ -5185,3 +5185,100 @@ if (isRandomAccessRange!Range && hasLength!Range)
          [1, 2, 0],
          [2, 1, 0]]));
 }
+/*
+Returns the array without any duplicates, keeping the initial order of appearance.
+If $(D T) is a basic type, a struct or a class which overrides the $(D toHash) operator
+the resulting array is computed in $(D O(n)) steps. If $(D T) is a class which does not
+override the $(D toHash) operator, the result is computed in $(D O(n^2)) steps. Aditionally,
+if the $(D opEquals) operator is not overriden for classes, $(D noDupes) considers the class
+pointer for duplication removal.
+*/
+T[] noDupes(T)(T[] s)
+{
+    static if (is(T == class) && !__traits(isOverrideFunction, T.toHash))
+    {
+        import std.algorithm : canFind;
+        T[] result;
+        foreach (T elem; s)
+            if (!result.canFind(elem))
+                result ~= elem;
+        return result;
+    }
+    else
+    {
+        bool[T] set;
+        T[] result;
+        foreach (T elem; s)
+            if (elem !in set)
+            {
+                set[elem] = true;
```

Comment:
I'm not sure if this strategy is something really clever with struct that are postblitable. The postblit will run once in the AA and then a second time in the result.